### PR TITLE
Beets: allow installing minimal version on legacy systems, fix dependencies

### PR DESCRIPTION
**Pull request by barracuda156**
#### Description

See commits.

Likely this is not gonna pass CI, since it requires a lot of ports, and uses python310.

###### Type(s)

- [x] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 10.6
Xcode 3.2

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL in commit message? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [ ] tried existing tests with `sudo port test`?
- [ ] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by barracuda156** [comment]
@reneeotten I am dropping py38 subport from `py-py7zr`, since it breaks CI due to some dependencies not building or not existing anymore. This is a nomaintainer port with no dependents for py38 version.

File: audio/beets/Portfile [hunk 1]
```diff
@@ -85,13 +85,20 @@ if {${name} eq ${subport} || "${name}-devel" eq ${subport}} {
...
                     port:ImageMagick \
+                    port:ffmpeg \
+                    port:flac \
+                    port:mp3val
+
+    # Do not pull in Go on PowerPC
+    if {${os.arch} ne "powerpc"} {
+        depends_run-append \
                     port:ipfs
+    }
```
Comment (reneeotten) [inline audio/beets/Portfile:101]: You are changing runtime/library dependencies here, so please increase the "revision".

**Review by reneeotten — CHANGES_REQUESTED** [review]

File: audio/beets/Portfile (hunk 1 above)
Comment (barracuda156) [reply audio/beets/Portfile:101]: Done

**Comment by reneeotten** [comment]
can you please rebase with `master` so we can see if the CI (or at least as much as possible) - right now it fails immediately with a parse error, which I assume has been resolved by now.

**Comment by barracuda156** [comment]
> can you please rebase with `master` so we can see if the CI (or at least as much as possible) - right now it fails immediately with a parse error, which I assume has been resolved by now.

Done

**Comment by barracuda156** [comment]
Ok, now it is just a missing dep in a subport:
```
  ERROR Missing dependencies:
  	build
  	importlib-metadata>=4.6
  Command failed:  cd "/opt/local/var/macports/build/_Users_runner_work_macports-ports_macports-ports_ports_python_py-py7zr/py39-py7zr/work/py7zr-0.22.0" && /opt/local/Library/Frameworks/Python.framework/Versions/3.9/bin/python3.9 -m build --no-isolation --wheel --outdir /opt/local/var/macports/build/_Users_runner_work_macports-ports_macports-ports_ports_python_py-py7zr/py39-py7zr/work 
  Error: Failed to build py39-py7zr: command execution failed
  DEBUG: Error code: CHILDSTATUS 93131 1
  DEBUG: Backtrace: command execution failed
      while executing
  "system {*}$notty {*}$callback {*}$nice $fullcmdstring"
      invoked from within
  "command_exec -callback portprogress::target_progress_callback build"
      (procedure "portbuild::build_main" line 10)
      invoked from within
  "$procedure $targetname"
  Exit code: 1
  Error: See /opt/local/var/macports/logs/_Users_runner_work_macports-ports_macports-ports_ports_python_py-py7zr/py39-py7zr/main.log for details.
  Error: Follow https://guide.macports.org/#project.tickets if you believe there is a bug.
  Error: Processing of port py39-py7zr failed
  Build of 'py39-py7zr' failed.
```

**Comment by barracuda156** [comment]
@reneeotten Anything left to be done here?

**Comment by barracuda156** [comment]
Ok, it suddenly decides it wants extra deps for py39 subport. I will try adding these.
```
  ERROR Missing dependencies:
  	build
  	importlib-metadata>=4.6
  Command failed:  cd "/opt/local/var/macports/build/_Users_runner_work_macports-ports_macports-ports_ports_python_py-py7zr/py39-py7zr/work/py7zr-0.22.0" && /opt/local/Library/Frameworks/Python.framework/Versions/3.9/bin/python3.9 -m build --no-isolation --wheel --outdir /opt/local/var/macports/build/_Users_runner_work_macports-ports_macports-ports_ports_python_py-py7zr/py39-py7zr/work 
  Error: Failed to build py39-py7zr: command execution failed
```

File: python/py-py7zr/Portfile [hunk 2]
```diff
@@ -20,25 +20,40 @@ checksums           rmd160  3aa2e3e10a71e3fe66f502fed601ba3d078b17a1 \
...
 if {${name} ne ${subport}} {
     depends_build-append \
                     port:py${python.version}-setuptools_scm
 
+    if {${python.version} < 310} {
+        depends_build-append \
+                    port:py${python.version}-build \
+                    port:py${python.version}-importlib-metadata
+    }
+
```
Comment (reneeotten) [inline python/py-py7zr/Portfile:34]:
```suggestion
```

**Comment by reneeotten** [comment]
@barracuda156 it shouldn't need the explicit dependencies added for the py39 subport. It builds fine for me locally in trace-mode. I'll merge this and we can keep an eye on the buildbot and see if the issue re-appears; I don't see anything wrong with it.

**Pull request merged**